kafka-spring-boot/boot-kafka-sample/src/test/java/com/example/springbootkafkasample/KafkaSampleIntegrationTest.java

```diff
@@ -45,27 +41,31 @@ class KafkaSampleIntegrationTest {
     @Order(1)
     void sendAndReceiveMessage() throws Exception {
         long initialCount = receiver2.getLatch().getCount();
-        this.mockMvc
-                .perform(post("/messages")
-                        .content(this.objectMapper.writeValueAsString(new MessageDTO("test_1", "junitTest")))
-                        .contentType(MediaType.APPLICATION_JSON))
-                .andExpect(status().isOk());
+        this.mockMvcTester
+                .post()
+                .uri("/messages")
+                .content(this.objectMapper.writeValueAsString(new MessageDTO("test_1", "junitTest")))
+                .contentType(MediaType.APPLICATION_JSON)
+                .assertThat()
+                .hasStatusOk();
 
         // 4 from topic1 and 3 from topic2 on startUp, plus 1 from test
         await().pollInterval(Duration.ofSeconds(1))
-                .atMost(Duration.ofSeconds(15))
+                .atMost(Duration.ofSeconds(30))
```

Comment:
@coderabbitai, Can you please help me to wait till the kafka status is changed to running and then initiated the polling?

Reply:
Hi @coderabbitai, above code checks for KafkaContainer which is part of `TestContainer` to check if it is running. Which springboot under the hood does it.

Since I let the application to create topics automatically it will be created lazily and here is status is not changed to RUNNING then junit is failing. I want to fix this issue.